syntax = "proto3";

package protos;
option go_package = "orch-go/gen/protos;protos";
option csharp_namespace = "Core";

import "custom_types.proto";
import "google/protobuf/empty.proto";
import "exchange_rate.proto";

service ExchangeRateAnalyticService {
  rpc ProcessGetAll (GetAllRequest) returns (GetAllExchangeRatesResponse);
  rpc ProcessGetById (GetExchangeRateByIdRequest) returns (ExchangeRateModel);
  rpc ProcessGetByIds (GetExchangeRateByIdsRequest) returns (GetAllExchangeRatesResponse);
  rpc ProcessGetByBaseCurrency (GetExchangeRateByBaseCurrencyRequest) returns (GetAllExchangeRatesResponse);
  rpc ProcessGetCount (google.protobuf.Empty) returns (CountResponse);
}
